engine/worker/cmd_key.go

```diff
@@ -142,27 +150,86 @@ func (wk *currentWorker) keyInstallHandler(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	wk.currentJob.pkey = path.Join(keysDirectory, key.Name)
+	switch key.Type {
+	case sdk.KeyTypeSSH:
+		wk.currentJob.pkey = path.Join(keysDirectory, key.Name)
+		if err := vcs.CleanSSHKeys(keysDirectory, nil); err != nil {
+			errClean := sdk.Error{
+				Message: fmt.Sprintf("Cannot clean ssh keys : %v", err),
+				Status:  http.StatusInternalServerError,
+			}
+			log.Error("%v", errClean)
+			writeJSON(w, errClean, errClean.Status)
+			return
+		}
 
-	if err := vcs.CleanSSHKeys(keysDirectory, nil); err != nil {
-		errClean := sdk.Error{
-			Message: fmt.Sprintf("Cannot clean ssh keys : %v", err),
-			Status:  http.StatusInternalServerError,
+		if err := vcs.SetupSSHKey(wk.currentJob.secrets, keysDirectory, key); err != nil {
+			errSetup := sdk.Error{
+				Message: fmt.Sprintf("Cannot setup ssh key %s : %v", keyName, err),
+				Status:  http.StatusInternalServerError,
+			}
+			log.Error("%v", errSetup)
+			writeJSON(w, errSetup, errSetup.Status)
+			return
 		}
-		log.Error("%v", errClean)
-		writeJSON(w, errClean, errClean.Status)
-		return
-	}
 
-	if err := vcs.SetupSSHKey(wk.currentJob.secrets, keysDirectory, key); err != nil {
-		errSetup := sdk.Error{
-			Message: fmt.Sprintf("Cannot setup ssh key %s : %v", keyName, err),
-			Status:  http.StatusInternalServerError,
+		writeJSON(w, keyResponse{PKey: wk.currentJob.pkey, Type: sdk.KeyTypeSSH}, http.StatusOK)
+
+	case sdk.KeyTypePGP:
+		content := []byte(key.Value)
+		tmpfile, errTmpFile := ioutil.TempFile("", key.Name)
+		if errTmpFile != nil {
+			errFile := sdk.Error{
+				Message: fmt.Sprintf("Cannot setup pgp key %s : %v", key.Name, errTmpFile),
+				Status:  http.StatusInternalServerError,
+			}
+			log.Error("%v", errFile)
+			writeJSON(w, errFile, errFile.Status)
+			return
+		}
+		defer func() {
+			_ = os.Remove(tmpfile.Name())
+		}()
+
+		if _, err := tmpfile.Write(content); err != nil {
+			errW := sdk.Error{
+				Message: fmt.Sprintf("Cannot setup pgp key file %s : %v", key.Name, err),
+				Status:  http.StatusInternalServerError,
+			}
+			log.Error("%v", errW)
+			writeJSON(w, errW, errW.Status)
+			return
+		}
+
+		if err := tmpfile.Close(); err != nil {
+			errC := sdk.Error{
+				Message: fmt.Sprintf("Cannot setup pgp key file %s (close) : %v", key.Name, err),
+				Status:  http.StatusInternalServerError,
+			}
+			log.Error("%v", errC)
+			writeJSON(w, errC, errC.Status)
+			return
 		}
-		log.Error("%v", errSetup)
-		writeJSON(w, errSetup, errSetup.Status)
-		return
-	}
 
-	writeJSON(w, keyResponse{PKey: wk.currentJob.pkey}, http.StatusOK)
+		cmd := exec.Command("gpg", "--import", tmpfile.Name())
```

Comment:
try to lookupPath to raise a proper error if there is no gpg
what about gpg2 on some debians ?